### ISSUE TITLE
Map: tiny countries should be visualized as circles

### ISSUE THREAD

**Issue by danielgavrilov** [owid/owid-grapher#1233]
## Core problem

Small countries are barely visible (or invisible) on our map chart. For example, `French Polynesia` is supposed to be shown in red on this map, but is invisible:

<img width="948" alt="Screenshot 2022-03-07 at 14 26 13" src="https://user-images.githubusercontent.com/1308115/157049596-ea842d7d-56cb-4c5f-b5ec-7f6618ce3482.png">

https://ourworldindata.org/explorers/migration-flows?Select+a+country=French+Polynesia&Outflow+or+Inflow=Immigrants:+Where+foreign-born+population+in+the+selected+country+moved+from&country=USA~GBR~FRA~MEX~IND~CHN~DEU~AUS~BRA~ZAF

## Proposed solution

Max suggests: Small countries should be visualized as solid circles (with some constant radius).

## Context

Slack thread: https://owid.slack.com/archives/C46U9LXRR/p1646656024971569

**Comment by danielgavrilov** [comment]
Discussing this with Daniel B: 
- If we do this project, it should be timeboxed. There is a big risk it takes a lot of time to get details right (e.g. circles not overlapping in the Caribbean)
- We should keep the same entities as in `MapTopology.ts`, no new territories/borders introduced

**Comment by danielgavrilov** [comment]
There was a variation at some point that implemented this, but it seems we decided against it: https://github.com/owid/owid-grapher/issues/83

**Comment by spoonerf** [comment]
I just realised that some small countries are not included on the [base map we use here](https://raw.githubusercontent.com/owid/owid-grapher/057ebc44df43557493ed0d364ed759d10651df4d/grapher/mapCharts/MapTopology.ts). For example Cayman Islands, Cape Verde, Curacao, Dominica, Mayotte... So including small circles for these would mean altering the `MapTopology.ts`, I think?

**Comment by covendev** [comment]
I would like to take up this issue.If it is open for external contributors

**Comment by larsyencken** [comment]
Hey @covendev! We aren't able to accept all contributions, since sometimes even if the feature looks visually right, it can take a lot of work to get it into our codebase conventions and into a shape we can maintain over time.

That said, if you have the time and interest, a proof-of-concept solution would still be useful to us, it would not be wasted.

**Comment by toni-sharpe** [comment]
@larsyencken not sure if you got a look, but I applied this on my own choropleth map, which started out as your co-ordinates, so in the spirit of "open-access/open-source" it's all available to re-use. It can be considered an exploration/POC of your problem but I very much owned it and did it my way, soi it's different. I also added quite a few countries and made my time slider handle country changes,

[The co-ordinates are here](https://github.com/toni-sharpe/data-visualisation--cdn/blob/main/WorldBorderList.min.json), it's now an array with manually placed labels (I had to do this to allow labels to be moved around and for different labels to be put on places that were in the same territory (Canaries) or one label for many (Marshall Isalnds): 

See here: 

### Overlapping islands are not such a big problem 
When zoom is avaialble (my map zooms from 1 .. 40) and hover working with the label too - here is the Carribean with a hover
<img width="1465" alt="Screenshot 2024-06-12 at 23 17 52" src="https://github.com/owid/owid-grapher/assets/10499070/a9366998-6bd6-4380-9cc2-78d336b656b1">

### Add zoom = 1, just dots are used
Zoom 1 is more about heat-maps over continents and regions really
<img width="1455" alt="Screenshot 2024-06-12 at 23 14 51" src="https://github.com/owid/owid-grapher/assets/10499070/b134a0b5-041d-432e-8953-4d1fb9b897c4">

### Anything that measures 1 X 1 cordinate size is a circle
But this doesn't work for places like Luxembourgh so a [hard-coded skip array deals with this](https://github.com/toni-sharpe/data-visualisation/blob/main/gui/src/util/Constant/BaseConstantList.js#L228-L235). This example at zoom = 8 shows several, some should be in the array, but the're actually not bad, arguably better as circles, afterall, this is a data/heat-map optimised tool, not a navigation tool, we've rather they were seen than accurately bordered
<img width="1099" alt="image" src="https://github.com/owid/owid-grapher/assets/10499070/863e66e3-3d6b-4998-807f-d013674dbc3a">

**Comment by danyx23** [comment]
@toni-sharpe nice to have this as a possible implementation strategy. For OWID we'll probably have to put in a bit of work to finally cretate a reproducible pipeline for creating the geojson artifact before we can tackle this, but when we get around to this we'll have a look at how you solved it to inform our take on it. Thanks!

**Comment by toni-sharpe** [comment]
@danyx23 happy to answer questions, just give me a flag and I'll see it in my Github notifications (which I have now brought into the realm of sanity after years of neglect with the intention of surfacing only things that matter and making it useful).

We could also connect on LinkedIn too: https://www.linkedin.com/in/toni-leigh-sharpe/ though my preference from an Open Source perspective is for discussions to take place on issues for all to read.

**Comment by larsyencken** [comment]
Thanks Toni, it's cool that you got it working on a larger map! As Daniel mentioned, there's a few dots to connect for us to convert it into a working Grapher, given how space constrained we are. The GeoJSON link you shared seems broken, could it be a private repo / Github rather than CDN link? You could also share in a gist if that's easier.

**Comment by toni-sharpe** [comment]
@larsyencken I needed to use the actual CDN link (which only has one node, but still, massive perf. boost)

The JSON should load from the link now.

**Comment by danyx23** [comment]
We are likely to add zoom to our maps next cycle. If we do, this issue becomes obsolete in our opinion.

**Comment by danyx23** [comment]
We now have zoom - this should be good enough and we don't plan on doing this anymore.

**Comment by spoonerf** [comment]
Hey @danyx23, 

Maybe it's more of a nice-to-have, but could we add small countries that are currently not on our map to the map? e.g. Martinique, Guadeloupe, Anguilla, British Virgin Islands, US Virgin Islands, Aruba, Cayman Islands, Curacao, Dominica, Mayotte... I'm sure there are more.

Now that we can zoom, their absence is a bit more obvious, and we do have data for many, if not all, of them.

**Comment by danyx23** [comment]
@spoonerf good point - I created #5275 as a follow-up issue for this concern!
